### PR TITLE
change: NewVar() to Var()

### LINKED ISSUE #4722: Scope::NewVar() not reflecting its implementation.
I think `NewVar` is not a appropriate name. When I first read the name: `NewVar`, I think it will create a new var, or fail. Not expecting that it will return a created var.

Here is the implementation of `NewVar`:

https://github.com/PaddlePaddle/Paddle/blob/c3bf332666bffb009a38bb568198e71673ff48d0/paddle/framework/scope.cc#L34-L43

`NewVar` does not create a new var every time, and does not return an error if no new var is created. `Var` seems to be a better name.

### LINKED ISSUE #4778: The semantic of Block_desc.Var() is not right
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/framework/block_desc.cc#L30

`Var()` should mean that if it contains this var, just return it. if it does not, then create and return it.

### PR DESCRIPTION
**Pull request by dzhwinter**
fix #4722 
fix #4778

### CODE REVIEW

File: paddle/framework/block_desc.cc [hunk 1]
```diff
@@ -18,7 +18,7 @@ limitations under the License. */
 namespace paddle {
 namespace framework {
 
-VarDescBind *BlockDescBind::NewVar(const std::string &name) {
+VarDescBind *BlockDescBind::GetOrCreateVar(const std::string &name) {
```
Comment (tonyyang-svail) [inline paddle/framework/block_desc.cc]: this should not me changed.

**Comment by helinwang** [comment]
@Superjom @dzhwinter From my understanding some of the current cpp implementations rely on `NewVar` returns a var if it's already created. Maybe we can create a new `NewVar` implementation that return false or raise exception when a var already exists. But change all current `NewVar` invocation to `GetOrCreateVar`, because currently no`NewVar` invocation is handling the error or exception and some of the `NewVar` invocation actually means to use `GetOrCreateVar`.

**Comment by Superjom** [comment]
I think, most of the current usage of `NewVar` just use it to `creates a new Variable` semantically. 
Better to change the `NewVar`'s implementation to `simply create new variable`, not just replace all the usages using a more higher level interface called `GetorCreateVar`. 
@helinwang

File: paddle/framework/block_desc.cc (hunk 1 above)
Comment (dzhwinter) [reply paddle/framework/block_desc.cc]: It's the same with `Scope`, also do the same logic of **find or create**.

**Comment by dzhwinter** [comment]
We should only provide one interface to create a variable. Otherwise, the user's code may be ugly mixed with `NewVar` and `GetOrCreateVar`.
Throw exception is also not good enough as well. It's not a big deal that people want to reuse a delared variable, so`GetOrCreateVar` cover the use case of `NewVar`.

**Comment by dzhwinter** [comment]
This PR depends #4728

File: paddle/framework/block_desc.cc [hunk 2]
```diff
@@ -18,7 +18,7 @@ limitations under the License. */
 namespace paddle {
 namespace framework {
 
-VarDescBind *BlockDescBind::NewVar(const std::string &name) {
+VarDescBind *BlockDescBind::Var(const std::string &name) {
   need_update_ = true;
   auto it = vars_.find(name);
   PADDLE_ENFORCE(it == vars_.end(), "Duplicated variable %s", name);
```
Comment (jacquesqiao) [inline paddle/framework/block_desc.cc]: var should mean getOrCreate

Comment (dzhwinter) [reply paddle/framework/block_desc.cc]: Done.

**Pull request merged**